packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -965,157 +969,167 @@ function ensureRootIsScheduled(root: FiberRoot, currentTime: number) {
 // This is the entry point for every concurrent task, i.e. anything that
 // goes through Scheduler.
 function performConcurrentWorkOnRoot(root, didTimeout) {
-  if (enableProfilerTimer && enableProfilerNestedUpdatePhase) {
-    resetNestedUpdateFlag();
+  if (enableFloat && supportsResources) {
+    prepareToRender(root.containerInfo);
```

Comment:
Search for `pushDispatcher` and `popDispatcher`. I think you can just call these in those functions. I believe they have the correct timing and you don't need to add another try.

Reply:
One thing we could support but hooks doesn’t is pre* functions I’m layout effects. It doesn’t quite make sense you would want to call it there necessarily but it is currently the phase that styles are inserted in and a preinit could in theory insert synchronously if called that way. One could then argue we should support in passive effects but currently we don’t. What was the argument to not support preinit in effects? I feel like we talked about this months ago but things have changed a lot since then

Reply:
It doesn't make sense to insert styles in the end of the layout phase, only if it's first. Once a layout effect has been called it might have read the layout which causes layout thrash if we change it after that.

Supporting it in effect is just like supporting it in event callbacks. So ensure you can start the download as early as possible. However, you get that with the "latest" doc approach anyway.

Reply:
ok convinced